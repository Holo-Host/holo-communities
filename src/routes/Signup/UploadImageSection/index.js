import React from 'react'
import { cameraSvg, loadingSvg } from 'util/assets'
import { bgImageStyle } from 'util/index'
import '../Signup.scss'

export default function UploadImage ({ avatarUrl, currentUser, updateSettingDirectly, loading }) {
  return null
}

export function uploadAvatar (currentUser, loading, avatarUrl) {
  let imageUrl = cameraSvg
  let styleName = 'upload-background-image'

  if (currentUser.avatarUrl) {
    imageUrl = currentUser.avatarUrl
    styleName = 'upload-background-image contain'
  }
  if (avatarUrl) {
    imageUrl = avatarUrl
    styleName = 'upload-background-image contain'
  }
  if (loading) {
    imageUrl = loadingSvg
    styleName = 'loading-background-image'
  }
  return <div styleName='image-upload-icon'>
    <div style={bgImageStyle(imageUrl)} styleName={styleName} />
  </div>
}
