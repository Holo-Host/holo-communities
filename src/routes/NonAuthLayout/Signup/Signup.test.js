import React from 'react'
import { shallow } from 'enzyme'
import Signup from './Signup'

describe('Signup', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Signup />)
    expect(wrapper).toMatchSnapshot()
  })
})
