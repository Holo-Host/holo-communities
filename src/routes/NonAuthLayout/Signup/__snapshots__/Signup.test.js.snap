// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup renders correctly 1`] = `
<div>
  <h1
    data-stylename="title"
  >
    Welcome to Hylo
  </h1>
  <p
    data-stylename="blurb"
  >
    Stay connected, organized, and engaged with your community.
  </p>
  <div
    data-stylename="field"
  >
    <label
      data-stylename="field-label"
      htmlFor="name"
    >
      Full name
    </label>
    <TextInput
      aria-label="name"
      autoFocus={true}
      inputRef={[Function]}
      label="name"
      name="name"
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    data-stylename="field"
  >
    <label
      data-stylename="field-label"
      htmlFor="email"
    >
      Your email address
    </label>
    <TextInput
      aria-label="email"
      inputRef={[Function]}
      label="email"
      name="email"
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    data-stylename="field"
  >
    <label
      data-stylename="field-label"
      htmlFor="password"
    >
      Password
    </label>
    <TextInput
      aria-label="password"
      label="password"
      name="password"
      onChange={[Function]}
      onEnter={[Function]}
      type="password"
    />
  </div>
  <Button
    data-stylename="submit"
    label="Sign Up"
    onClick={[Function]}
  />
  <p
    data-stylename="connect-label"
  >
    Or connect with:
  </p>
  <div
    data-stylename="auth-buttons"
  >
    <a
      aria-label="Log in with Facebook"
      data-stylename="facebook"
      onClick={[Function]}
      tabIndex={0}
    >
      <Icon
        data-stylename="auth-icon"
        name="Facebook"
      />
      Facebook
    </a>
    <a
      aria-label="Log in with Google"
      data-stylename="google"
      onClick={[Function]}
      tabIndex={0}
    >
      <Icon
        data-stylename="auth-icon"
        name="Google"
      />
      Google
    </a>
  </div>
</div>
`;

